exporters/stdout/stdoutlog/exporter_test.go

```diff
@@ -0,0 +1,228 @@
+
+	// Flush the exporter
+	err = exporter.ForceFlush(context.Background())
+	assert.NoError(t, err)
+
+	// Shutdown the exporter
+	err = exporter.Shutdown(context.Background())
+	assert.NoError(t, err)
+
+	// Export a record after shutdown, this should not be written
+	err = exporter.Export(context.Background(), []sdklog.Record{record})
+	assert.NoError(t, err)
+
+	// Check the writer
+	assert.Equal(t, expectedJSON(now, false), buf.String())
+}
+
+func TestExporterExport(t *testing.T) {
+	now := time.Now()
+
+	record := getRecord(now)
+	records := []sdklog.Record{record, record}
+
+	// Get canceled context
+	canceledCtx, cancel := context.WithCancel(context.Background())
+	cancel()
+
+	testCases := []struct {
+		name           string
+		options        []stdoutlog.Option
+		ctx            context.Context
+		records        []sdklog.Record
+		expectedResult string
+		expectedError  error
+	}{
+		{
+			name:           "default",
+			options:        []stdoutlog.Option{},
+			ctx:            context.Background(),
+			records:        records,
+			expectedResult: expectedJSONs(now, false),
+		},
+		{
+			name:           "NoRecords",
+			options:        []stdoutlog.Option{},
+			ctx:            context.Background(),
+			records:        nil,
+			expectedResult: "",
+		},
+		{
+			name:           "WithPrettyPrint",
+			options:        []stdoutlog.Option{stdoutlog.WithPrettyPrint()},
+			ctx:            context.Background(),
+			records:        records,
+			expectedResult: expectedJSONs(now, true),
+		},
+		{
+			name:           "WithoutTimestamps",
+			options:        []stdoutlog.Option{stdoutlog.WithoutTimestamps()},
+			ctx:            context.Background(),
+			records:        records,
+			expectedResult: expectedJSONs(time.Time{}, false),
+		},
+		{
+			name:           "WithoutTimestamps and WithPrettyPrint",
+			options:        []stdoutlog.Option{stdoutlog.WithoutTimestamps(), stdoutlog.WithPrettyPrint()},
+			ctx:            context.Background(),
+			records:        records,
+			expectedResult: expectedJSONs(time.Time{}, true),
+		},
+		{
+			name:           "WithCanceledContext",
+			ctx:            canceledCtx,
+			records:        records,
+			expectedResult: "",
+			expectedError:  context.Canceled,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Write to buffer for testing
+			var buf bytes.Buffer
+
+			exporter, err := stdoutlog.New(append(tc.options, stdoutlog.WithWriter(&buf))...)
+			assert.NoError(t, err)
+
+			err = exporter.Export(tc.ctx, tc.records)
+			assert.Equal(t, tc.expectedError, err)
+			assert.Equal(t, tc.expectedResult, buf.String())
+		})
+	}
+}
+
+func expectedJSONs(now time.Time, prettyPrint bool) string {
+	result := expectedJSON(now, prettyPrint)
+	return result + result
+}
+
+// revive:disable-next-line:flag-parameter
+func expectedJSON(now time.Time, prettyPrint bool) string {
+	serializedNow, _ := json.Marshal(now)
+	if prettyPrint {
```

Comment:
Can we avoid assertion logic and e.g.  have two functions instead like `expectedJSON` and `expectedPrettyJSON`? 

Or even better refactor it to "approval tests" (also known as "golden master" technique) so that can even have the expected output as files in `testdata`. The `expectedResult` can be the name of the file in `testdata`.

References: 
- http://xunitpatterns.com/Conditional%20Test%20Logic.html
- https://quii.gitbook.io/learn-go-with-tests/go-fundamentals/html-templates#introducing-approval-tests (let's not add any dependency)